File: ios_app/iphone/ui/Utils/pxDp.swift
import UIKit

let H_PADDING = 16.0
let H_PADDING_HALF = H_PADDING / 2

private let scale: CGFloat = UIScreen.main.scale
let onePx: CGFloat = 1 / scale
let halfDpFloor: CGFloat = CGFloat(Int(scale / 2)) / scale // -=
let halfDpCeil: CGFloat = 1 - halfDpFloor // +=

File: android/src/main/java/me/timeto/app/ui/px_dp.kt
package me.timeto.app.ui

import android.content.res.Resources
import androidx.compose.ui.unit.dp

private val density: Float = Resources.getSystem().displayMetrics.density
fun dpToPx(dp: Float): Int = (dp * density).toInt()
fun pxToDp(px: Int): Float = (px / density)

val H_PADDING = 16.dp
val H_PADDING_HALF = H_PADDING / 2
val onePx = pxToDp(1).dp

val halfDpFloor = pxToDp(dpToPx(1f) / 2).dp // -=
val halfDpCeil = 1.dp - halfDpFloor // +=
